### PR TITLE
CIF-2177 - Expose configured product identifiers for components

### PR DESCRIPTION
**Pull request by dplaton**
## Description

Add a `data-product-sku` attribute to the DOM of the following components:
* Product Teaser (will also update the Venia custom component)
* Product Collection
* Product
* Product Carousel

The Sling Models for the above components all expose the SKU in some way or another, this data attribute just renders that information.

## Related Issue

CIF-2177

## Motivation and Context

To properly implement the "Associated Commerce Content" side-panel in the editor all components that render a product must expose the SKU in a data attribute. The Associated Commerce Content side-panel will show the assets tagged with that SKU.

## How Has This Been Tested?

Functional testing, Selenium ITs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java [hunk 1]
```diff
@@ -61,23 +61,32 @@
...
-     * Returns the SKU of the product displayed by this {@code ProductTeaser}
-     * 
+     * Returns the SKU of the product displayed by this {@code ProductTeaser}. If the teaser displays a variant, then this method returns
+     * the SKU of the variant.
+     *
      * @return a String value representing the SKU
      */
     String getSku();
 
+    /**
```
Comment (Buuhuu) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java:71]: Why is this needed? The information is present in the sku and the format is also kown/fixed to `sku#variant` in many places. Can the consumer extract the "master product sku" instead from the sku and we save the redundancy here?

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/package-info.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
-@Version("3.0.0")
+@Version("4.0.0")
```
Comment (Buuhuu) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/package-info.java:16]: When removing the masterproductsku this is should not be necessary anymore.

**Review by Buuhuu — CHANGES_REQUESTED** [review]

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java (hunk 1 above)
Comment (dplaton) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java:71]: The problem here is that the Product Teaser does not expose the `sku#variant` format. Also, no consumer is extracting the "master product SKU", everything is done at the model level. 

Some context: all the components that display products rely on the "master SKU" because you cannot search by variant SKU in Magento. The product teaser is the only one (apart from the `Product` component, but that's a special case) that retrieves the product directly by the SKU, so that's why it doesn't expose the `combinedSku` format.

Comment (Buuhuu) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java:71]: If any other place returns `sku#variant`, why doesn't the product teaser? We should fix that instead. Afaics only the data layer currently consumes this sku. 

I am -1 introducing this technical dept in combination with a major interface change. 

Lets better introduce a productteaser v2 that does the sku right.

Comment (dplaton) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java:71]: > If any other place returns sku#variant

No other places return `sku#variant`. All the models return the master product SKU, that's why I didn't have to make any changes in other models.

**Review by herzog31 — APPROVED** [review]

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html [hunk 3]
```diff
@@ -1,30 +1,29 @@
 <!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-  ~ Copyright 2019 Adobe
```
Comment (Buuhuu) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html:2]: Please don't reformat the license header. They were recently changed, adapting the format from the WCM Core Components

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html [hunk 4]
```diff
@@ -1,30 +1,29 @@
...
-        </a>
-    </div>
+      <div role="displaycard"
+           id="${product.id}"
+           data-cmp-data-layer="${product.data.json}"
+           class="card product__card"
+           data-product-sku="${product.SKU}">
+          <a href="${product.URL}">
+              <div class="product__card__image">
+                  <img class="product__image" src="${product.imageURL}" data-gallery-role="currentimage"/>
```
Comment (Buuhuu) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html:22]: Is there any particular reason why you removed the `alt`?

Comment (dplaton) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html:22]: I didn't, I think I didn't pay attention to a merge operation.
